core/math/random_pcg.h

```diff
@@ -38,18 +38,18 @@
 class RandomPCG {
 	pcg32_random_t pcg;
 	uint64_t current_seed; // seed with this to get the same state
+	uint64_t current_inc;
 
 public:
 	static const uint64_t DEFAULT_SEED = 12047754176567800795U;
 	static const uint64_t DEFAULT_INC = PCG_DEFAULT_INC_64;
 	static const uint64_t RANDOM_MAX = 0xFFFFFFFF;
 
-	RandomPCG(uint64_t p_seed = DEFAULT_SEED, uint64_t p_inc = PCG_DEFAULT_INC_64);
+	RandomPCG(uint64_t p_seed = DEFAULT_SEED, uint64_t p_inc = DEFAULT_INC);
 
 	_FORCE_INLINE_ void seed(uint64_t p_seed) {
 		current_seed = p_seed;
-		pcg.state = p_seed;
-		pcg32_random_r(&pcg); // Force changing internal state to avoid initial 0
+		pcg32_srandom_r(&pcg, current_seed, current_inc);
```

Comment:
From http://www.pcg-random.org/using-pcg-c-basic.html:

> For this generator, there are 2^63 possible sequences of pseudorandom numbers. Each sequence is entirely distinct and has a period of 2^64. The initseq argument selects which stream you will use. The initstate argument specifies where you are in that 2^64 period.

Maybe `current_inc` should be renamed `current_seq` to better reflect it's usage? Actual `inc` from pcg32_srandom_r is `(initseq << 1u) | 1u` so this might be a bit confusing.

Reply:
Its also required to change `pcg32_random_t `cuz its have `inc` inside. I think its redundant, and its not a target for this PR